packages/fx-core/src/component/debugHandler/utils.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+"use strict";
+
+import {
+  CryptoProvider,
+  LogProvider,
+  M365TokenProvider,
+  ProjectSettingsV3,
+  ResourceContextV3,
+  TelemetryReporter,
+  UserInteraction,
+  v3,
+  err,
+  ok,
+  InputsWithProjectPath,
+  Platform,
+  Result,
+  Void,
+  FxError,
+} from "@microsoft/teamsfx-api";
+
+import { ComponentNames } from "../constants";
+import { DefaultManifestProvider } from "../resource/appManifest/manifestProvider";
+import { checkWhetherLocalDebugM365TenantMatches } from "../../plugins/solution/fx-solution/v2/utils";
+import { isCSharpProject } from "../utils";
+
+export async function checkM365Tenant(
+  projectPath: string,
+  projectSettingsV3: ProjectSettingsV3,
+  envInfoV3: v3.EnvInfoV3,
+  m365TokenProvider: M365TokenProvider,
+  logger: LogProvider,
+  telemetry: TelemetryReporter,
+  ui: UserInteraction,
+  cryptoProvider: CryptoProvider
+): Promise<Result<Void, FxError>> {
+  const resourceContextV3 = constructResourceContextV3(
+    projectPath,
+    projectSettingsV3,
+    envInfoV3,
+    m365TokenProvider,
+    logger,
+    telemetry,
+    ui,
+    cryptoProvider
+  );
+  const inputs = constructInputsWithProjectPath(projectPath);
+  const tenantId =
+    envInfoV3.state[ComponentNames.AadApp]?.tenantId ||
+    envInfoV3.state[ComponentNames.AppManifest]?.tenantId;
+  const checkResult = await checkWhetherLocalDebugM365TenantMatches(
```

Comment:
Is there the case - user does not login, and run the task(s) directly. Will this ask user to login again and again?

Reply:
`checkWhetherLocalDebugM365TenantMatches` does ask user to login if user has not loged in.  The tasks are running in sequence, so only the first time login is required, and the following steps will run as normal.

Reply:
I mean, what if use cancels login? Or that's out of scope?

Reply:
If user cancels login, there will be an error to terminate the running task.
![image](https://user-images.githubusercontent.com/37978464/196356458-3f51b135-955c-4ab9-a840-5b873e02f143.png)